server/utils/auth/command_gate.py

```
@@ -142,6 +142,62 @@ def gate_command(
+            return
+        org_id = getattr(state, "org_id", None)
+        if not org_id:
+            return
+        dirty_key = f"tool_perms_dirty:{org_id}"
+        if not rc.get(dirty_key):
+            return
+        from utils.db.connection_pool import db_pool
+        from utils.auth.stateless_auth import set_rls_context
+        user_id = getattr(state, "user_id", None)
+        with db_pool.get_connection() as conn:
+            with conn.cursor() as cur:
```

Ruff (0.15.12)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)